File: src/main/java/javalin/embeddedserver/EmbeddedServer.java
// Javalin - https://javalin.io
// Copyright 2017 David Åse
// Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE

package javalin.embeddedserver;

public interface EmbeddedServer {

    int start(String host, int port) throws Exception;

    void join() throws InterruptedException;

    void stop();

    int activeThreadCount();

    Object attribute(String key);
}

File: src/main/java/io/javalin/embeddedserver/EmbeddedServer.kt
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin.embeddedserver

/**
 * Defines the underlying Javalin server.
 */
interface EmbeddedServer {

    @Throws(Exception::class)
    fun start(port: Int): Int

    @Throws(Exception::class)
    fun stop()

    fun activeThreadCount(): Int
    fun attribute(key: String): Any

}
